syntax = "proto3";
package game;

service Game {
    rpc StartGame (StartGameRequest) returns (StartGameResponse);

    rpc PlayGame (PlayGameRequest) returns (PlayGameResponse);
}

message StartGameRequest {
    Size world_size = 1;
}

message StartGameResponse {
    int32 game_id = 1;
    Map world_map = 2;
}

message PlayGameRequest {
    string message = 1;
}

message PlayGameResponse {
    string message = 1;
}

message Size {
    int32 width = 1;
    int32 height = 2;
}

message Map {
    enum Cell {
        Empty = 0;
        Wall = 1;
    };
    Size map_size = 1;
    repeated Cell cells = 2;
}
